### PR TITLE
[Codegen][Tuner] Allow tuning specs in the LLVMGPU pipeline

### PR DESCRIPTION
**Pull request by kuhar**
This adds the `materialize-tuning-specs` pass to the LLVMGPU executable configuration pipelines.

Add a test that shows that the tuning spec gets applied and picked up in the ROCDL pipeline.

Also, replace the print-based checks in existing tests with op remarks on transform strategy application in `materialize-user-configs`.

Issue: https://github.com/iree-org/iree/issues/19214

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp [hunk 1]
```diff
@@ -194,6 +201,9 @@ struct MaterializeUserConfigsPass final
       //      ```
       LDBG("MaterializeUserConfigsPass on function: " << funcOp);
       if (succeeded(userTransformLibrary)) {
+        StringRef libraryModuleName =
+            userTransformLibrary->transformLibrary.getSymName().value_or(
+                "<unnamed>");
```
Comment (Groverkss) [inline compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp:206]: nit: braces

File: compiler/src/iree/compiler/Codegen/Common/test/materialize_user_config_from_tuning_spec.mlir [hunk 2]
```diff
@@ -25,16 +24,14 @@
...
-// PARENT-NEXT:   func.func @main_0
-//
 // PARENT-LABEL:  module @parent attributes {
 // PARENT-SAME:     iree_codegen.tuning_spec_mlirbc = dense<
 // PARENT-LABEL:    module @child {
 // PARENT:            func.func @main_0
 
 module @parent {
   module @child {
+    // expected-remark@+1 {{Applied transform configuration strategy @iree_linked_tuning_spec::@__kernel_config}}
```
Comment (Groverkss) [inline compiler/src/iree/compiler/Codegen/Common/test/materialize_user_config_from_tuning_spec.mlir:34]: this is much nicer

**Review by Groverkss — APPROVED** [review]
LGTM

File: compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp (hunk 1 above)
Comment (kuhar) [reply compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp:206]: ? the whole block is already in braces. I think you might have misread the indentation in this `if`

File: compiler/src/iree/compiler/Codegen/Common/test/materialize_user_config_from_tuning_spec.mlir (hunk 2 above)
Comment (kuhar) [reply compiler/src/iree/compiler/Codegen/Common/test/materialize_user_config_from_tuning_spec.mlir:34]: thanks for the suggestion

File: compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp [hunk 3]
```diff
@@ -1183,6 +1183,7 @@ static void buildLLVMGPUCodegenConfigurationPassPipelineImpl(
     funcPassManager.addPass(createConfigTrackingCanonicalizerPass);
     funcPassManager.addPass(createCSEPass);
   }
+  modulePassManager.addPass(createMaterializeTuningSpecsPass());
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp:1186]: I think the pass ordering might be off... Do we want to do user configs first and then tuning spec. User config should be the ultimate override.

Comment (kuhar) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp:1186]: User configs do come first. It's documented in the materialize user configs pass. Materialize tuning specs doesn't run the transform dialect interpreter and it's up to the materialize user configs to decide which transforms library to run.

Comment (kuhar) [reply compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp:1186]: Maybe I need to come up with a different name to better disambiguate these two passes...

**Pull request merged**